cellpack/autopack/Gradient.py

```diff
@@ -176,13 +198,16 @@ def pick_point_for_ingredient(ingr, allIngrPts, all_gradients):
         if isinstance(ingr.gradient, list):
             if len(ingr.gradient) > 1:
                 if not hasattr(ingr, "combined_weight"):
-                    gradient_list = [
-                        gradient
-                        for gradient_name, gradient in all_gradients.items()
-                        if gradient_name in ingr.gradient
-                    ]
+                    gradient_list = []
+                    for gradient_name in ingr.gradient:
+                        if gradient_name not in all_gradients:
+                            raise ValueError(
+                                f"Gradient {gradient_name} not found in gradient list"
+                            )
+                        gradient_list.append(all_gradients[gradient_name])
+
                     combined_weight = Gradient.get_combined_gradient_weight(
-                        gradient_list
+                        gradient_list, ingr.gradient_weights
```

Comment:
here might pass an empty `gradient_list` to `get_combined_gradient_weight` and result in an `IndexError`. Maybe add a check in `get_combined_gradient_weight`

Reply:
In this case `gradient_list` should not be empty, since we check that `ingr.gradient` is a list with more than one element. All elements from `ingr.gradient` are added to `gradient_list` and an error is raised if there is a missing entry. I'll add a check to `get_combined_gradient_weight` just to make sure this is the case.

Reply:
oh, that makes sense! Thank you for the clarification and adding the check